src/compiler/transformers/update-component-class.ts

```diff
@@ -50,17 +50,25 @@ const createConstClass = (
   const constModifiers: ts.Modifier[] = [];
 
   if (transformOpts.componentExport !== 'customelement') {
-    constModifiers.push(ts.createModifier(ts.SyntaxKind.ExportKeyword));
+    constModifiers.push(ts.factory.createModifier(ts.SyntaxKind.ExportKeyword));
   }
 
-  return ts.createVariableStatement(
+  return ts.factory.createVariableStatement(
     constModifiers,
-    ts.createVariableDeclarationList(
+    ts.factory.createVariableDeclarationList(
       [
-        ts.createVariableDeclaration(
+        ts.factory.createVariableDeclaration(
           className,
           undefined,
-          ts.createClassExpression(classModifiers, undefined, classNode.typeParameters, heritageClauses, members)
+          undefined,
```

Comment:
We add `undefined` here because the fn sig changes when we move to the `factory` version
```diff
- createClassExpression: (modifiers: readonly Modifier[] | undefined, name: string | Identifier | undefined, typeParameters: readonly TypeParameterDeclaration[] | undefined, heritageClauses: readonly HeritageClause[] | undefined, members: readonly ClassElement[]) => ClassExpression;
+ createClassExpression: (decorators: readonly Decorator[] | undefined, modifiers: readonly Modifier[] | undefined, name: string | Identifier | undefined, typeParameters: readonly TypeParameterDeclaration[] | undefined, heritageClauses: readonly HeritageClause[] | undefined, members: readonly ClassElement[]): ClassExpression;
```